tests/ct_out_of_memory_handling.c

```diff
@@ -1,20 +1,24 @@
+#include "adapters/libevent.h"
 #include "hircluster.h"
 #include "test_utils.h"
 #include <assert.h>
+#include <stdbool.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 
 #define CLUSTER_NODE "127.0.0.1:7000"
 
 int successfulAllocations = 0;
+bool assertWhenAllocFail = false; // Enable for troubleshooting
```

Comment:
Strictly what you actually mean is to *make an assertion to fail when alloc fails*, so maybe `failAssertionWhenAllocFails` but can't it be done with a macro? Then it can be controlled with -D to the compiler.

How about something like...

```C
#ifdef STOP_ON_OOM
#define assertWhenAllocFail() { bool alloc_success = false; assert(alloc_success); }
#else
#define assertWhenAllocFail()
#endif 
```

Reply:
This has been useful when troubleshooting a specific OOM trigger-point, i.e to enable it just for a specific part of the testcase, likebefore a call that indirect triggers a memleak/crash. Its not very useful to set it for all since it will only stop on the first malloc.